packages/dashboard/src/DashboardPlugin.ts

```diff
@@ -84,3 +84,29 @@ export interface DashboardPlugin {
   /** Called when the dashboard is unintialized and layout is about to be destroyed */
   deinitialize?: (config: DashboardConfig) => void;
 }
+
+/**
+ * Takes a partial DashboardPluginComponentProps and verifies all the dashboard component fields are filled in.
+ * @param props The props to check
+ * @returns True if the props are valid DashboardPluginComponentProps, false otherwise
+ */
+export function isDashboardPluginProps(
+  props: Partial<DashboardPluginComponentProps>
+): props is DashboardPluginComponentProps {
+  return (
+    props.id !== undefined &&
+    props.layout !== undefined &&
+    props.panelManager !== undefined &&
+    props.registerComponent !== undefined
```

Comment:
Can any of these be more specific? `typeof id !== 'string'`? `typeof registerComponent !== 'function'`?

Reply:
Yes